source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -267,8 +293,12 @@ void Filter::populateRateLimitDescriptors(const Router::RateLimitPolicy& rate_li
             fmt::format("ratelimit.{}.http_filter_enabled", disable_key), 100)) {
       continue;
     }
-    rate_limit.populateDescriptors(descriptors, config_->localInfo().clusterName(), headers,
-                                   callbacks_->streamInfo());
+    const bool apply_on_stream_done = rate_limit.applyOnStreamDone();
+    has_apply_on_stream_done_policy_ |= apply_on_stream_done;
+    if ((on_stream_done && apply_on_stream_done) || (!on_stream_done && !apply_on_stream_done)) {
```

Comment:
I think simple `^` or (on_stream_done == apply_on_stream_done) should be fine.

Reply:
well nice catch...